include/boost/geometry/algorithms/detail/is_valid/has_invalid_coordinate.hpp

```diff
@@ -80,26 +80,26 @@ struct indexed_has_invalid_coordinate
 
 struct range_has_invalid_coordinate
 {
+private:
     struct point_has_valid_coordinates
     {
         template <typename Point>
-        static inline bool apply(Point const& point)
+        inline bool operator()(Point const& point)
         {
-            return ! point_has_invalid_coordinate::apply(point);
+            return point_has_invalid_coordinate::apply(point);
         }
     };
 
+public:
     template <typename Geometry, typename VisitPolicy>
     static inline bool apply(Geometry const& geometry, VisitPolicy& visitor)
     {
         boost::ignore_unused(visitor);
 
-        bool const has_valid_coordinates = detail::check_iterator_range
-            <
-                point_has_valid_coordinates,
-                true // do not consider an empty range as problematic
-            >::apply(geometry::points_begin(geometry),
-                     geometry::points_end(geometry));
+        bool const has_valid_coordinates = geometry::points_end(geometry) == 
+            std::find_if(geometry::points_begin(geometry),
+                         geometry::points_end(geometry),
+                         point_has_valid_coordinates());
```

Comment:
This is a search for points with invalid coordinates.

If not found then `true` is returned, also for empty range, ok.